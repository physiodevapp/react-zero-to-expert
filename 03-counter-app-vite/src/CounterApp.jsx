import { useState } from 'react';
import PropTypes from 'prop-types';

function CounterApp({ value }) {

  // console.log('render');

  const [count, setCount] = useState(value)

  const handleCounter = (key) => {
    // setCount(count + 1)
    switch (key) {
      case 1:
        setCount((prevCount) => prevCount + 1);
        break;
      case -1:
        setCount((prevCount) => prevCount - 1);
      break;
      default:
        setCount(value);
        break;
    }
  }

  return (
    <>
      <h1>Counter App</h1>
      <h2>{count}</h2>
      <button onClick={() => handleCounter(1)}>
        +1
      </button>
      <button onClick={() => handleCounter(-1)}>
        -1
      </button>
      <button onClick={() => handleCounter()}>
        Reset
      </button>
    </>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
  value: 0
}

export default CounterApp