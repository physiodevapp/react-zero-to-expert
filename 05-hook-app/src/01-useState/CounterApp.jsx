import React, { useState } from "react";

function CounterApp() {
  const [counters, setCounters] = useState({
    counter1: 10,
    counter2: 20,
    counter3: 30,
  });

  const { counter1, counter2, counter3 } = counters

  const handleclick = () => {
    setCounters(() => {
      return { 
        ...counters,
        counter1: counter1 + 1,
      }
    })
  }

  return (
    <>
      <h1>Counter: {counter1}</h1>
      <h1>Counter: {counter2}</h1>
      <h1>Counter: {counter3}</h1>

      <hr />

      <button className="btn btn-primary" onClick={handleclick}>+1</button>
    </>
  );
}

export default CounterApp;
