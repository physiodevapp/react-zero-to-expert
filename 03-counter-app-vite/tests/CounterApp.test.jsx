import { render, screen } from "@testing-library/react";
import CounterApp from "../src/CounterApp";

describe("Testing <CounterApp/>", () => {
  const initValue = 100;

  test("should match snapshot", () => {
    const { container } = render(<CounterApp value={initValue} />);

    expect(container).toMatchSnapshot();
  });

  test("should show intial value", () => {
    render(<CounterApp value={initValue} />);

    expect(screen.getByRole("heading", { level: 2 }).innerHTML).toBe(initValue.toString());
  });
});
