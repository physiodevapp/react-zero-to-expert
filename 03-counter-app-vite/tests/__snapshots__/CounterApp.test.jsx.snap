// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <CounterApp/> should match snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    100
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button>
    Reset
  </button>
</div>
`;
